src/components/Dropdown/Dropdown.tsx

```diff
@@ -222,48 +229,93 @@ export default class Dropdown extends AutoControlledComponent<
             getRootProps,
             getToggleButtonProps,
             isOpen,
+            toggleMenu,
             highlightedIndex,
             selectItemAtIndex,
           }) => {
+            const accessibilityRootProps = getRootProps(
+              { refKey: 'innerRef' },
+              { suppressRefError: true },
+            )
+            const { innerRef, ...accessibilityRootPropsRest } = accessibilityRootProps
             return (
-              <div
-                className={classes.containerDiv}
-                onClick={this.handleContainerClick.bind(this, isOpen)}
-              >
-                {multiple && this.renderSelectedItems(styles)}
-                {search &&
-                  this.renderSearchInput(
-                    getRootProps,
-                    getInputProps,
+              <Ref innerRef={innerRef}>
+                <div
+                  className={classes.container}
+                  onClick={multiple ? this.handleContainerClick.bind(this, isOpen) : undefined}
+                >
+                  {multiple && this.renderSelectedItems(styles)}
+                  {search
+                    ? this.renderSearchInput(
+                        accessibilityRootPropsRest,
+                        getInputProps,
+                        highlightedIndex,
+                        selectItemAtIndex,
+                        variables,
+                      )
+                    : this.renderTriggerButton(styles, getToggleButtonProps)}
+                  {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
+                  {this.renderItemsList(
+                    styles,
+                    variables,
+                    isOpen,
                     highlightedIndex,
+                    toggleMenu,
                     selectItemAtIndex,
+                    getMenuProps,
+                    getItemProps,
+                    getInputProps,
                   )}
-                {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
-                {this.renderItemsList(
-                  styles,
-                  variables,
-                  getMenuProps,
-                  getItemProps,
-                  isOpen,
-                  highlightedIndex,
-                )}
-              </div>
+                </div>
+              </Ref>
             )
           }}
         </Downshift>
       </ElementType>
     )
   }
 
+  private renderTriggerButton(
+    styles: ComponentSlotStylesInput,
+    getToggleButtonProps: (options?: GetToggleButtonPropsOptions) => any,
+  ): JSX.Element {
+    const { placeholder, itemToString, multiple } = this.props
+    const { value } = this.state
+    const content = multiple ? placeholder : value ? itemToString(value) : placeholder
```

Comment:
💣 

Can we simplify this somehow? May be to a separate function.

```jsx
const computeContentForButton = ({ placeholder, itemToString, multiple }: Props, { value }: State) => {
  if (multiple) return placeholder 
  return value ? itemToString(value) : placeholder
}
```

Reply:
is it hard to read? creating a function that will be used once? Inside this method? Destructuring again? is it really worth it?

Reply:
Yep, it's hard to read because it requires more attention.

https://eslint.org/docs/rules/no-nested-ternary
https://github.com/airbnb/javascript/blob/eslint-config-airbnb-v17.1.0/packages/eslint-config-airbnb-base/rules/style.js#L323

BTW, this rule will be enabled in #600.